libkbfs/rekey_fsm.go

```diff
@@ -437,6 +437,15 @@ func (m *rekeyFSM) loop() {
 	for {
 		select {
 		case e := <-m.reqs:
+			if e.eventType == rekeyShutdownEvent {
+				// Set m.reqs to nil so on next iteration, we will skip any
+				// content in m.reqs. So if there are multiple
+				// rekeyShutdownEvent, we won't close m.shutdownCh multiple
+				// times.
+				m.reqs = nil
```

Comment:
I'm not sure this will work the way you want it to; if another goroutine tries to send to `m.reqs` at the same time, it seems like there would be a race there. `close` is atomic, assignment is not.